### PR TITLE
feat(lambda): add `ipv6AllowedForDualStack` for lambda function

### PR DESCRIPTION
**Pull request by scanlonp**
This property sets the cloudformation `VpcConfigProperty` property of the same name. This feature will enable Lambda functions to be configured to allow or disallow IPv6 traffic when using dual-stack VPC configurations.

Related #894.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by scanlonp** [comment]
@Mergifyio update

**Comment by scanlonp** [comment]
@Mergifyio update

**Comment by mikewrighton** [comment]
Can you link the issue hwew as `Fixes #...`?

**Comment by scanlonp** [comment]
@mikewrighton there is no specific issue for this cfn coverage gap. I put related to the larger ipv6 issue.

File: packages/@aws-cdk-testing/framework-integ/test/aws-lambda/test/integ.vpc-lambda.ts [hunk 1]
```diff
@@ -14,4 +15,14 @@ new lambda.Function(stack, 'MyLambda', {
   vpc,
 });
 
-app.synth();
+new lambda.Function(stack, 'IPv6EnabledLambda', {
```
Comment (Vandita2020) [inline packages/@aws-cdk-testing/framework-integ/test/aws-lambda/test/integ.vpc-lambda.ts:18]: I tried the existing PR and its integration tests were failing, here's my fork with changes to integration tests and almost similar changes for the feature implementation that works: https://github.com/Vandita2020/aws-cdk/commit/e2c7753bc9d80196d1c3e757efb627193d80f1eb

**Comment by scanlonp** [comment]
#28928 adds this functionality.

**Pull request closed without merging**